storage2/src/storage.rs

```diff
@@ -3,10 +3,14 @@ use std::{path::PathBuf, sync::Arc};
 use anyhow::Result;
 use rocksdb::DB;
 
+use crate::snapshot::Snapshot;
 use crate::State;
 
-#[derive(Clone, Debug)]
-pub struct Storage(Arc<DB>);
+#[derive(Clone)]
+pub struct Storage {
+    snapshot: Arc<Snapshot>,
```

Comment:
Maybe this should be called `latest_snapshot`, for clarity?  Also, will we need to wrap this in a `parking_lot::RwLock` so that we can insert the new latest snapshot?